app/models/target.rb

```diff
@@ -71,4 +71,24 @@ def to_s
     parts = [geography.name, target_type&.humanize, year]
     parts.compact.join(' - ')
   end
+
+  def as_json(options)
+    path = options[:path]
+    {
+      id: id,
+      iso_code3: geography.iso,
+      doc_type: source,
+      type: target_type,
+      scope: scopes.map(&:name).join(', '),
+      sector: sector&.name,
+      description: description,
+      sources: legislations.map do |l|
+        {
+          id: l.id,
+          title: l.title,
+          link: "#{path}/cclow/geographies/#{geography.id}/#{l.route}/#{id}"
```

Comment:
Rails.application.routes.url_helpers could help with taht

Reply:
gonna move this to the controller.